ui-speedspacechart/src/types/chartTypes.ts

```diff
@@ -93,6 +93,20 @@ export type DrawFunctionParams = {
   setStore?: React.Dispatch<React.SetStateAction<Store>>;
 };
 
+export type SpeedLimitTagsLayerDrawingStore = Pick<
+  Store,
+  'speedLimitTags' | 'ratioX' | 'leftOffset' | 'speeds'
+> & {
+  layersDisplay: Pick<Store['layersDisplay'], 'electricalProfiles' | 'powerRestrictions'>;
```

Comment:
Question: do we really need to optimize the case where the user updates the visibility of a layer other than these two?

Just wondering if it's really worth the more involved type and picking.

Reply:
Honestly not really, but I still did this in order to avoid having non primary types in the useEffect dependency array